gateway-service/src/main/java/org/zowe/apiml/gateway/ribbon/ApimlRibbonRetryFactory.java

```diff
@@ -10,57 +10,67 @@
 
 package org.zowe.apiml.gateway.ribbon;
 
-import lombok.RequiredArgsConstructor;
-import lombok.experimental.Delegate;
 import org.springframework.cloud.client.loadbalancer.LoadBalancedRetryContext;
 import org.springframework.cloud.client.loadbalancer.LoadBalancedRetryPolicy;
 import org.springframework.cloud.client.loadbalancer.ServiceInstanceChooser;
 import org.springframework.cloud.netflix.ribbon.RibbonLoadBalancedRetryFactory;
+import org.springframework.cloud.netflix.ribbon.RibbonLoadBalancedRetryPolicy;
+import org.springframework.cloud.netflix.ribbon.RibbonLoadBalancerContext;
 import org.springframework.cloud.netflix.ribbon.SpringClientFactory;
+import org.springframework.http.HttpMethod;
 import org.springframework.retry.RetryListener;
 
+import java.lang.reflect.InvocationTargetException;
+import java.net.ConnectException;
 import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Allows adding RetryListeners to Ribbon Retry
  */
 public class ApimlRibbonRetryFactory extends RibbonLoadBalancedRetryFactory {
 
+    private final SpringClientFactory clientFactory;
     private final AtomicReference<ServiceInstanceChooser> serviceInstanceChooser = new AtomicReference<>();
 
     public ApimlRibbonRetryFactory(SpringClientFactory clientFactory) {
         super(clientFactory);
+        this.clientFactory = clientFactory;
     }
 
     @Override
     public LoadBalancedRetryPolicy createRetryPolicy(String service, ServiceInstanceChooser serviceInstanceChooser) {
         this.serviceInstanceChooser.set(serviceInstanceChooser);
-        return new LoadBalancedRetryPolicyFix(super.createRetryPolicy(service, serviceInstanceChooser));
-    }
+        RibbonLoadBalancerContext lbContext = this.clientFactory.getLoadBalancerContext(service);
+        return new RibbonLoadBalancedRetryPolicy(service, lbContext, serviceInstanceChooser, clientFactory.getClientConfig(service)) {
 
-    @Override
-    public RetryListener[] createRetryListeners(String service) {
-        return new RetryListener[] {
-            new InitializingRetryListener(this.serviceInstanceChooser.get()),
-            new AbortingRetryListener()
-        };
-    }
+            @Override
+            public boolean canRetry(LoadBalancedRetryContext context) {
+                HttpMethod method = context.getRequest().getMethod();
+                return HttpMethod.GET == method || lbContext.isOkToRetryOnAllOperations() || isConnectionRefused(context.getLastThrowable());
```

Comment:
Since you mentioned it, why copying of the condition instead of `super.canRetry() || isConnectionRefused(context.getLastThrowable())` ?

Reply:
Because it would always retry for all connection errors even if the retry limit is reached.

Reply:
Actually you are right :-)